rspack.config.ts

```diff
@@ -0,0 +1,786 @@
+const HAS_WEBPACK_DEV_SERVER_CONFIG =
+  !!SENTRY_BACKEND_PORT && !!SENTRY_WEBPACK_PROXY_PORT;
+
+// User/tooling configurable environment variables
+const NO_DEV_SERVER = !!env.NO_DEV_SERVER; // Do not run webpack dev server
+const SHOULD_FORK_TS = DEV_MODE && !env.NO_TS_FORK; // Do not run fork-ts plugin (or if not dev env)
+const SHOULD_HOT_MODULE_RELOAD = DEV_MODE && !!env.SENTRY_UI_HOT_RELOAD;
+const SHOULD_ADD_RSDOCTOR = Boolean(env.RSDOCTOR);
+
+// Storybook related flag configuration
+const STORYBOOK_TYPES = Boolean(env.STORYBOOK_TYPES) || IS_PRODUCTION;
+
+// Deploy previews are built using vercel. We can check if we're in vercel's
+// build process by checking the existence of the PULL_REQUEST env var.
+const DEPLOY_PREVIEW_CONFIG = IS_DEPLOY_PREVIEW && {
+  branch: env.NOW_GITHUB_COMMIT_REF,
+  commitSha: env.NOW_GITHUB_COMMIT_SHA,
+  githubOrg: env.NOW_GITHUB_COMMIT_ORG,
+  githubRepo: env.NOW_GITHUB_COMMIT_REPO,
+};
+
+// When deploy previews are enabled always enable experimental SPA mode --
+// deploy previews are served standalone. Otherwise fallback to the environment
+// configuration.
+const SENTRY_EXPERIMENTAL_SPA =
+  !DEPLOY_PREVIEW_CONFIG && !IS_UI_DEV_ONLY ? !!env.SENTRY_EXPERIMENTAL_SPA : true;
+
+// We should only read from the SENTRY_SPA_DSN env variable if SENTRY_EXPERIMENTAL_SPA
+// is true. This is to make sure we can validate that the experimental SPA mode is
+// working properly.
+const SENTRY_SPA_DSN = SENTRY_EXPERIMENTAL_SPA ? env.SENTRY_SPA_DSN : undefined;
+const CODECOV_TOKEN = env.CODECOV_TOKEN;
+// value should come back as either 'true' or 'false' or undefined
+const ENABLE_CODECOV_BA = env.CODECOV_ENABLE_BA === 'true';
+
+// this is the path to the django "sentry" app, we output the webpack build here to `dist`
+// so that `django collectstatic` and so that we can serve the post-webpack bundles
+const sentryDjangoAppPath = path.join(__dirname, 'src/sentry/static/sentry');
+const distPath = path.join(sentryDjangoAppPath, 'dist');
+const staticPrefix = path.join(__dirname, 'static');
+
+// Locale compilation and optimizations.
+//
+// Locales are code-split from the app and vendor chunk into separate chunks
+// that will be loaded by layout.html depending on the users configured locale.
+//
+// Code splitting happens using the splitChunks plugin, configured under the
+// `optimization` key of the webpack module. We create chunk (cache) groups for
+// each of our supported locales and extract the PO files and moment.js locale
+// files into each chunk.
+//
+// A plugin is used to remove the locale chunks from the app entry's chunk
+// dependency list, so that our compiled bundle does not expect that *all*
+// locale chunks must be loaded
+const localeCatalogPath = path.join(
+  __dirname,
+  'src',
+  'sentry',
+  'locale',
+  'catalogs.json'
+);
+
+type LocaleCatalog = {
+  supported_locales: string[];
+};
+
+const localeCatalog: LocaleCatalog = JSON.parse(
```

Comment:
Outside of the scope of this PR, but it would be interesting if we could time this locale procedure to see how much it contributes to the startup cost.